File: 507.go
package leetcode

import "math"

func checkPerfectNumber(num int) bool {
	if num == 1 {
		return false
	}
	sum := 1
	for i := 2; i < int(math.Sqrt(float64(num)))+1; i++ {
		if num%i == 0 {
			sum += i
			if tmp := num / i; tmp != i {
				sum += tmp
			}
		}
	}
	return sum == num
}

File: 507.py
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        if num == 1:
            return False
        s = 1
        for i in range(2, int(num**0.5)+1):
            if num % i == 0:
                s += i
                if (tmp := num // i) != i:
                    s += tmp
        return s == num
